src/Microsoft.ML.Data/Prediction/CalibratorCatalog.cs

```diff
@@ -49,39 +48,35 @@ public interface ICalibrator
         where TICalibrator : class, ICalibrator
     {
-        protected readonly IHostEnvironment Host;
-        protected readonly TCalibratorTrainer CalibratorTrainer;
-
-        protected readonly IPredictor Predictor;
-        protected readonly SchemaShape.Column ScoreColumn;
-        protected readonly SchemaShape.Column FeatureColumn;
-        protected readonly SchemaShape.Column LabelColumn;
-        protected readonly SchemaShape.Column WeightColumn;
-        protected readonly SchemaShape.Column PredictedLabel;
-
-        protected CalibratorEstimatorBase(IHostEnvironment env,
-            TCalibratorTrainer calibratorTrainer,
-            IPredictor predictor = null,
-            string labelColumn = DefaultColumnNames.Label,
-            string featureColumn = DefaultColumnNames.Features,
-            string weightColumn = null)
+        [BestFriend]
+        private protected readonly IHostEnvironment Host;
+        private readonly ICalibratorTrainer _calibratorTrainer;
+        ICalibratorTrainer IHaveCalibratorTrainer.CalibratorTrainer => _calibratorTrainer;
+
+        [BestFriend]
+        private protected readonly SchemaShape.Column ScoreColumn;
+        [BestFriend]
+        private protected readonly SchemaShape.Column LabelColumn;
+        [BestFriend]
+        private protected readonly SchemaShape.Column WeightColumn;
+        [BestFriend]
+        private protected readonly SchemaShape.Column PredictedLabel;
+
+        [BestFriend]
+        private protected CalibratorEstimatorBase(IHostEnvironment env,
+            ICalibratorTrainer calibratorTrainer, string labelColumn, string scoreColumn, string weightColumn)
         {
             Host = env;
-            Predictor = predictor;
-            CalibratorTrainer = calibratorTrainer;
+            _calibratorTrainer = calibratorTrainer;
 
-            ScoreColumn = TrainerUtils.MakeR4ScalarColumn(DefaultColumnNames.Score); // Do we fantom this being named anything else (renaming column)? Complete metadata?
-            LabelColumn = TrainerUtils.MakeBoolScalarLabel(labelColumn);
-            FeatureColumn = TrainerUtils.MakeR4VecFeature(featureColumn);
-            PredictedLabel = new SchemaShape.Column(DefaultColumnNames.PredictedLabel,
-                SchemaShape.Column.VectorKind.Scalar,
-                BoolType.Instance,
-                false,
-                new SchemaShape(MetadataUtils.GetTrainerOutputMetadata()));
+            if (!string.IsNullOrWhiteSpace(labelColumn))
+                LabelColumn = TrainerUtils.MakeBoolScalarLabel(labelColumn);
+            else
+                env.CheckParam(!calibratorTrainer.NeedsTraining, nameof(labelColumn), "For trained calibrators, " + nameof(labelColumn) + " must be specified.");
+            ScoreColumn = TrainerUtils.MakeR4ScalarColumn(scoreColumn); // Do we fanthom this being named anything else (renaming column)? Complete metadata?
```

Comment:
>// Do we fanthom this being named anything else (renaming column)? Complete metadata? [](start = 72, length = 85)

want to keep this?

Reply:
You mean the comment? I didn't quite know what the original author had in mind, so I just fixed the obvious typo and moved on.